sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CallerInformationParametersShouldBeLast.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Method1(string callerFilePath) { }
+        public void Method2([CallerFilePath]string callerFilePath) { }
+        public void Method3(string other, [CallerFilePath]string callerFilePath) { }
+        public void Method4([CallerFilePath]string callerFilePath, string other) { } // Noncompliant {{Move 'callerFilePath' to the end of the parameter list.}}
+//                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        public void Method5([CallerFilePath]string callerFilePath, string other, [CallerLineNumber]int callerLineNumber) { }
+//                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+        public void Method6(string first, [CallerFilePath]string callerFilePath, [CallerLineNumber]int callerLineNumber, string other) { }

```

Comment:
Cannot decide whether it is better to report once (and display the different parameters) or multiple times.

Reply:
Hhhmm. If we reported one diagnostic with all of the params, the squigglie would only appear under one of them even if we supplied a list of _AdditionalLocations_, wouldn't it?

The UX might be a bit odd if we only reported one issue against multiple params - as soon as the user fixes one, the number of issues won't change but the squigglie will jump to another location.
They'd also be unable to suppress the issue for just one parameter (although it's difficult to think of a scenario in which they'd want to do that).

I'd say report multiple issues. It's simpler.

Reply:
Fine by me.